flink-runtime/src/test/java/org/apache/flink/runtime/filecache/FileCacheReadsFromBlobTest.java

```
@@ -0,0 +1,140 @@
+		+ "deine Boten, Herr, verehren Das sanfte Wandeln deines Tags.";
+
+	@Rule
+	public final TemporaryFolder temporaryFolder = new TemporaryFolder();
+
+	private FileCache fileCache;
+
+	@Mock
+	private PermanentBlobService blobService;
+
+	@Before
+	public void setup() throws IOException {
+		try {
+			String[] tmpDirectories = new String[]{temporaryFolder.newFolder().getAbsolutePath()};
+			fileCache = new FileCache(tmpDirectories, blobService);
+		} catch (Exception e) {
+			e.printStackTrace();
+			fail("Cannot create FileCache: " + e.getMessage());
+		}
+	}
+
+	@After
+	public void shutdown() {
+		try {
+			fileCache.shutdown();
+		} catch (Exception e) {
+			e.printStackTrace();
+			fail("FileCache shutdown failed: " + e.getMessage());
+		}
+	}
+
+	@Test
+	public void testFileDownloadedFromBlob() {
+		try {
+			JobID jobID = new JobID();
+
+			final PermanentBlobKey permanentBlobKey = new PermanentBlobKey();
+			when(blobService.getFile(jobID, permanentBlobKey)).thenAnswer(inv -> {
+					File f = temporaryFolder.newFile("cacheFile");
+					try {
+						Files.write(testFileContent, f, Charsets.UTF_8);
+					} catch (Exception e) {
+						e.printStackTrace();
+						fail("Error initializing the test: " + e.getMessage());
+					}
+					return f;
+				}
+			);
+
+			final String fileName = "test_file";
+			// copy / create the file
+			final DistributedCache.DistributedCacheEntry entry = new DistributedCache.DistributedCacheEntry(
+				fileName,
+				false,
+				true,
+				InstantiationUtil.serializeObject(permanentBlobKey));
+			Future<Path> copyResult = fileCache.createTmpFile(fileName, entry, jobID);
+
+			final Path dstPath = copyResult.get();
+			final String actualContent =
+				StringUtils.join(Files.readLines(new File(dstPath.toUri()), StandardCharsets.UTF_8), "\n");
+			assertTrue(dstPath.getFileSystem().exists(dstPath));
+			assertEquals(testFileContent, actualContent);
+
+		} catch (Exception e) {
```

just let the exception fail the test, i.e. no catching.